### ISSUE TITLE
Yolo26 Pose errors on DDP + compile="default"

### ISSUE THREAD

**Issue by ExtReMLapin** [ultralytics/ultralytics#24408]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Title.

`self.flow_model = RealNVP()` in `Pose26` is never used in `forward()` so it raises an error with compile

Sorry for very short issue but I'm out of bandwidth and the line above is more than enough to understand what's going on. ☮️ 

### Environment

Doesn't matter, it's on HEAD, 3 GPUS

### Minimal Reproducible Example

Literally just train on compile="default" with multiple gpu and pose

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ExtReMLapin, thank you for reporting this potential YOLO26 pose training bug with DDP and `compile="default"` 🚀 This is an automated response to help gather the details needed for investigation, and an Ultralytics engineer will also assist you soon.

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and related training references.

Since this appears to be a 🐛 Bug Report, please provide a full [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Your summary is helpful, but for DDP/compile issues we still need the exact training command or script, model variant, dataset config, complete error traceback, and package/version details so we can reproduce the behavior reliably on our side 🔍

If this is related to custom training behavior, please also share as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify this is not already resolved on the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello.

Thanks for reporting. I can reproduce it.

**Comment by ExtReMLapin** [comment]
perfect !

**Comment by mulhamfetna** [comment]
I've worked on DDP and torch.compile compatibility issues.

**Technical Analysis:**
YOLO26 Pose fails with DDP + compile suggests:
1. Dynamic shapes issue with Pose head
2. Non-traceable control flow in Pose model
3. Static_graph configuration needed for compile + DDP

**Root Cause (likely):**
- torch.compile doesn't handle all Pose-specific operations
- DDP adds complexity with gradient synchronization
- Combined effect causes model execution errors

**Fix Strategy:**
1. Add static_graph=True for compile + DDP
2. Identify non-compilable operations in Pose head
3. Add guards for dynamic tensor shapes
4. Consider torch._dynamo.config settings

Related to #24410 (broadcast_buffers) which is being addressed.

**Comment by mulhamfetna** [comment]
Created fix PR: molhamfetnah/ultralytics#3

The fix adds  to DDP when  is enabled. This allows DDP to handle the variable graph structure that occurs with YOLO26 Pose models (RealNVP flow, keypoint branches).

**Changes:**
- : Add  when both DDP and compile are used

**Comment by ExtReMLapin** [comment]
Fucking hell, since claude code, all I can see if an army of indians and pakis spamming pull requests for fixes that are already proposed.

Go in their history and they have nothing before agentic coding existed

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has now been merged in [PR #24413: `Use static_graph=True for compile and remove find_unused_parameters`](https://github.com/ultralytics/ultralytics/pull/24413).

This change may resolve the issue by improving DDP behavior in multi-GPU training, especially when using `torch.compile` and in edge cases where a batch contains no ground-truth labels. The main updates include:

- Switching DDP setup from `find_unused_parameters=True` to `static_graph=bool(self.args.compile)` in `trainer.py`
- Adding a safeguard so `denoising_class_embed` is still treated as used during training when a batch has zero labels
- Improving compatibility for cases where parameters may be reused multiple times in a forward pass under compiled training

Credit to @Y-T-G for the PR, with contributions from @Laughing-q and @ExtReMLapin.

If you'd like to test the fix now, you can install the latest `main` branch with:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Or, if preferred, wait for the next official PyPI release.

Please let us know whether these changes resolve the issue in your setup. Thanks 🙏 for reporting this, and further feedback is very welcome if the issue persists.
